lib/blocklist.c

```diff
@@ -102,8 +104,33 @@ void allowlist_prefix(char *ip, int prefix_len)
 	_add_constraint(addr, prefix_len, ADDR_ALLOWED);
 }
 
+static int is_ip_ipv6(char *ip) {
+	// don't modify the input string
+	int length = strlen(ip);
+	char *new_str = (char*)malloc(length);
+	if (new_str == NULL) {
+		log_fatal("blocklist", "could not allocate memory for IPv6 validation");
+	}
+	strncpy(new_str, ip, length);
```

Comment:
There's an off by one here. `The strlen() function returns the number of characters that precede the terminating NUL character.`. However, `new_str` is defined as the length of the string without the null byte at the end. Last, `strncpy` is pretty much deprecated and we should use `strlcpy`.

Reply:
Last, I'll just note that `strndup` is a function and I suspect that's what you're looking for :)